### ISSUE TITLE
Support different casing

### ISSUE THREAD

**Issue by gdoron** [jitbit/MapDataReader#2]
First and foremost, thanks a lot for your surely hard work on this project! Greatly appreciated.

For almost all databases the columns naming convention is snake_case, but with MapDataReader it won't map back to the entity if it's in C#'s naming convention - PascalCase.

What do you think about changing `GetDeterministicHashCode` behavior to remove underscores when looking for a match in addition to ignore casing differences?

https://github.com/jitbit/MapDataReader/blob/1948ca25dea3ad8adde6b7ac07e9e430eed94486/MapDataReader/MapperGenerator.cs#L116-L134

Maybe it can be a flag on `GenerateDataReaderMapperAttribute`?

**Comment by gdoron** [comment]
@alex-jitbit
Update: Personally for me, I went a different route because how horrible snowflake's SDK for dotnet is.
I had to do many modifications that required a lot of manual casting and serialization.

This FR is not important for me anymore, but I believe for others it 100% would.

Thanks a lot of the library, looks awesome❣️

**Comment by alex-jitbit** [comment]
Currently the workaround would be to either use DTO class or SQL aliases

`SELECT my_colymn AS MyProperty`

I'll consider make it case-insensitive

**Comment by alex-jitbit** [comment]
Actually, it is already case-insensitive, fixed in d59601e1763c43786346e5b38162b3ccc1717982

As for the "removing underscores" - I believe it would be an easter egg. Try using aliases in your query like described above

**Comment by gdoron** [comment]
Unforturently I couldn't use a different DTO or aliases to my queries because I can hot-swap the database on request, so exactly the same SQL query should behave no matter the database.
Only for the snowflake provider I had to go down the datareader level.

And since the interface is simply `List<T> GetData(string sql)`, I can't really use a different DTO either.
Like I updated earlier for me, it's not that important anymore as we took a different path, but I believe it's a very important feature to have, so I wouldn't close as completed but your call.

Cheers mate!
